recipes.c

```diff
@@ -656,61 +656,193 @@ int checkRecipe(struct ItemCombination combo, int *makeableItems, int *outputsCr
  *			  struct Recipe	   *recipeList
  * Outputs	: 1 if we can still make all remaining recipes with the
  *			  current inventory. Else, return 0
- * 
+ *
  * This function iterates over all recipes, skips over any that have
  * been already created, and calls checkRecipe to see if each remaining
  * recipe can still be fulfilled at some point in the roadmap.
  -------------------------------------------------------------------*/
-int stateOK(struct Inventory inventory, int *outputsCreated, struct Recipe *recipeList) {
+int stateOK(struct Inventory inventory, const int * const outputsCreated, struct Recipe *recipeList) {
 	// With the given inventory, can the remaining recipes be fulfilled?
-		
+
 	// If Chapter 5 has not been done, verify that Thunder Rage is in the inventory
 	if (!outputsCreated[getIndexOfRecipe(Dried_Bouquet)] && indexOfItemInInventory(inventory, Thunder_Rage) == -1) {
 		return 0;
 	}
-	
+
 	int makeableItems[NUM_ITEMS] = {0};
 
 	placeInventoryInMakeableItems(makeableItems, inventory);
-	
+
 	// If Chapter 5 has not been done, add the items it gives
 	if (outputsCreated[getIndexOfRecipe(Dried_Bouquet)] == 0) {
 		makeableItems[Keel_Mango] = 1;
 		makeableItems[Coconut] = 1;
-		makeableItems[Dried_Bouquet] = 1;
+	 	makeableItems[Dried_Bouquet] = 1;
 		makeableItems[Courage_Shell] = 1;
 	}
-	
+
 	// List of items to not try to make
 	// Once we're done exploring the current recipe, unset it in the array
-	int dependentRecipes[NUM_RECIPES] = {0};
-	
+	static int dependentRecipes[NUM_RECIPES] = {0};
```

Comment:
Are you sure that making this static is safe? The program is multithreaded.

Reply:
Oh, oops. Nope, that's not remotely safe.

Reply:
Just un-static'd it in commit https://github.com/SevenChords/CipesAtHome/pull/16/commits/9b38868133a06f4216ae94c8800e50dbb477799b